apps/control-plane/cmd/server/grant_routes_test.go

```
@@ -0,0 +1,200 @@
+				w.WriteHeader(http.StatusOK)
+			})
+
+			authzMW := authz.NewMiddleware(func(r *http.Request) (authz.Actor, error) {
+				return tc.caller.actor, nil
+			})
+
+			mux := http.NewServeMux()
+			registerCreditGrantRoutes(mux, stubAuth(tc.caller), authzMW, adminSpy, selfSpy)
+
+			rec := httptest.NewRecorder()
+			mux.ServeHTTP(rec, httptest.NewRequest(tc.method, tc.path, nil))
```

golangci-lint (2.12.2)

[error] net/http/httptest.NewRequest must not be called. use net/http/httptest.NewRequestWithContext

(noctx)